gphys/gdalet.F

```diff
@@ -0,0 +1,161 @@
+c------------------------------------------------------------------------------
+	real function vdmform(x,rmas,isw) ! VDM formfactor
+	real*4 x, rmas, dalit, rhom	  ! masses in MeV
+        real*4 f, delta
+	parameter (rhom=770.0)
+C
+	vdmform = 0.0
+	if (x.ge.rmas) return
+        if (x.lt.2.0*0.511) return
+
+        f = sqrt(1.-4.*(0.511/x)**2) * (1.+2.*(0.511/x)**2)
+        if (isw.eq.1) then       ! pseudo-scalar meson -> gamma e+ e-
+	   dalit = ((1.-(x/rmas)**2)**3/(x/rmas))*1./(1.-(x/rhom)**2)**2
+	   vdmform = 8.2e-4*dalit*f
+        else if (isw.eq.2) then  ! vector meson -> pi0 e+ e-
+           delta = (135.0/rmas)**2
+           bracket = (1.+(x/rmas)**2/(1.-delta))**2
+     +                - 4.*(x/rmas)**2/(1.-delta)**2
+           if(bracket.le.0.0) return
+           ff2 = 0.64**4/((0.65**2-(x/1000.)**2)**2 + (0.042*0.65)**2)
+           dalit = (bracket**1.5)/(x/rmas)*ff2
+           vdmform = 8.2e-4*dalit*f
+        end if
+C
+        return
+	end
+c------------------------------------------------------------------------------
+	real function qedform(x,rmas,isw) ! QED formfactor
+	real*4 x, rmas, dalit		  ! masses in MeV
```

Comment:
Please add description of the function arguments. It is also helpful to add reference to the paper where the form-factors are taken from.